framework-plugins/lisk-framework-http-api-plugin/test/functional/delegates.spec.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright © 2020 Lisk Foundation
+ *
+ * See the LICENSE file at the top-level directory of this distribution
+ * for licensing information.
+ *
+ * Unless otherwise agreed in a custom licensing agreement with the Lisk Foundation,
+ * no part of this software, including this file, may be copied, modified,
+ * propagated, or distributed except according to the terms contained in the
+ * LICENSE file.
+ *
+ * Removal or modification of this copyright notice is prohibited.
+ */
+import { Application } from 'lisk-framework';
+import axios from 'axios';
+import { callNetwork, createApplication, closeApplication, getURL } from './utils/application';
+
+describe('Delegates endpoint', () => {
+	let app: Application;
+	const firstDelegateAccount = {
+		address: 'A/bZC329BJfcOlLRwn4ju4x1iX8=',
+		balance: '0',
+		nonce: '0',
+		keys: { numberOfSignatures: 0, mandatoryKeys: [], optionalKeys: [] },
+		asset: {
+			delegate: {
+				username: 'genesis_34',
+				pomHeights: [],
+				consecutiveMissedBlocks: 0,
+				lastForgedHeight: 0,
+				isBanned: false,
+				totalVotesReceived: '1000000000000',
+			},
+			sentVotes: [
+				{
+					delegateAddress: 'A/bZC329BJfcOlLRwn4ju4x1iX8=',
+					amount: '1000000000000',
+				},
+			],
+			unlocking: [],
+		},
+	};
+
+	beforeAll(async () => {
+		app = await createApplication('delegates_http_functional');
+	});
+
+	afterAll(async () => {
+		await closeApplication(app);
+	});
+
+	describe('/api/delegates', () => {
+		it('should respond with all the delegates', async () => {
```

Comment:
A suggestion, I prefer to group tests in terms of response codes, which match the right expectations and structure if you see in the swagger file. 

```yaml
responses:
        200:
          description: Registered delegate accounts.
          schema:
            $ref: '#/definitions/DelegateResponse'
        400:
          description: Malformed query or parameters.
          schema:
            $ref: '#/definitions/ErrorResponse'
        429:
          description: Too many requests, exceeded rate limit.
          schema:
            $ref: '#/definitions/ErrorResponse'
        500:
          description: Unexpected error.
          schema:
            $ref: '#/definitions/ErrorResponse'
```

So if we write expectations also grouped in same way that looks easy to read and track.